icicle_v3/backend/cpu/src/curve/cpu_msm.hpp

```diff
@@ -0,0 +1,667 @@
+#pragma once
+
+#include <thread>
+#include <atomic>
+#include <string>
+#include <iostream>
+#include <fstream>
+
+#include "icicle/errors.h"
+#include "icicle/runtime.h"
+#include "icicle/config_extension.h"
+#include "icicle/curves/projective.h"
+#include "icicle/curves/curve_config.h"
+#include "icicle/msm.h"
+#include "tasks_manager.h"
+
+using namespace icicle;
+
+#ifdef DUMMY_TYPES // for testing
+using A = DummyP;
+using P = DummyP;
+using scalar_t = DummyScalar;
+#else
+using A = curve_config::affine_t;
```

Comment:
it now seems confusing that both those and the templates use A,P. Also, it doesn't support G2. 
Now that everything is template I think you can move this logic to the test. It would be better than here.